pkg/vm/vm.go

```diff
@@ -842,14 +836,11 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 		v.estack.PushVal(new(big.Int).Xor(b, a))
 
 	case opcode.EQUAL, opcode.NOTEQUAL:
-		b := v.estack.Pop()
-		if b == nil {
-			panic("no top-level element found")
+		if v.estack.Len() < 2 {
+			panic("need a pair of elements on the stack")
 		}
+		b := v.estack.Pop()
 		a := v.estack.Pop()
```

Comment:
As I remember, the order of pops and checks directly affects what is on stack in case of fault invocation.
I wonder if this is a problem now (with the exception handling).

Reply:
Exceptions are not a problem, they're thrown/catched explicitly. But stack contents in case of failure can differ, that's true. I'm not sure we're 100% compliant with C# for that now.